### PR TITLE
BotService 0.1.8: update ARM templates, use async zipdeploy

### PR DESCRIPTION
**Pull request by stevengum**
- Update `az bot publish` to use zipdeploy API asynchronously
- Fix bug in example command output from `az bot create`
- Update v4 SDK ARM template to include `"Microsoft-BotFramework-AppId"`, `"Microsoft-BotFramework-AppPassword"` and `"SCM_DO_BUILD_DURING_DEPLOYMENT"`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by stevengum** [comment]
Still need to pick up/merge Yugang's changes for AutoRest4; I'm currently rerecording tests right now. 

@tjprescott sorry for the late submission but it would be fantastic if this PR could make it in today!

**Comment by stevengum** [comment]
@yugangw-msft, @williexu friendly ping for review 😄

File: src/command_modules/azure-cli-botservice/HISTORY.rst [hunk 1]
```diff
@@ -5,6 +5,10 @@ Release History
 
 0.1.7
 +++++
+* Add "SCM_DO_BUILD_DURING_DEPLOYMENT" to ARM template's Application Settings for v4 Web App Bots
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-botservice/HISTORY.rst]: This should be put on top of `0.1.7`, as the `0.1.7` has been released

Comment (stevengum) [reply src/command_modules/azure-cli-botservice/HISTORY.rst]: Sorry about that! Fixed 😄

**Comment by stevengum** [comment]
@yugangw-msft I have to rerecord the tests per some feedback from other members on the Bot Framework team (ARM template changes).

I've updated the HISTORY.rst per your comments and will wait for this Travis build to finish to see if any static check or other linting errors pop up. After that I'll push the new recordings ASAP.

**Comment by stevengum** [comment]
@yugangw-msft, sorry for the delay, the TravisCI build is currently running

**Comment by hansmbakker** [comment]
Does this PR break publishing when publishing a webapp bot that was provisioned without the `SCM_DO_BUILD_DURING_DEPLOYMENT` setting?

I'm getting warnings in my Azure DevOps pipeline since recently:
` WARNING: Didn't detect SCM_DO_BUILD_DURING_DEPLOYMENT or its value was "false" in App Service's Application Settings. Build may not commence during deployment. To learn how to trigger a build when deploying to your App Service, see https://github.com/projectkudu/kudu/wiki/Deploying-from-a-zip-file`

**Pull request merged**